src/libAtomVM/opcodesswitch.h

```diff
@@ -995,7 +1181,11 @@ static bool maybe_call_native(Context *ctx, AtomString module_name, AtomString f
 
     unsigned int i = 0;
 
+    int current_label = -1;
```

Comment:
I tried to find a scenario where the value would not be correct and would refer to another process. It's not obvious that the current_label is correct when an error occurs, even if it seems to be the case. Maybe some opcodes, such as OP_APPLY, can generate bad values. I am wondering if this souldn't be in the context, but then the performance penalty can be higher.

Reply:
I really need your advice here, especially around the SMP branch.  Would this local variable be "scheduler-safe"?

Reply:
SMP scheduler would probably not change things except by making a possible bug random.

The issue here is that this code is relying on a property of the compiled code which, AFAIK, is not clearly specified. This property is that the code is always scheduled in to code positions where no exception can occur before a label opcode, in such a way that the current_label of a scheduled out process is overwritten by the scheduled in process by a label opcode before being read by the RAISE_ERROR macro.

I'm not completely certain this is true. I tried a little bit to prove it is wrong with some wierd (non-SMP) test code without success.

If this properly does not hold, you will need to process the current label the same way the instruction pointer is processed, by saving it in the context on schedule out and restoring it on schedule in.

Alternatively, as discussed earlier, you could save the instruction offset instead of the label and only get the label (when necessary) by using another data structure (so does BEAM AFAIK). I don't know if you could even take advantage of yet another feature of the compiler that labels are (apparently) monotonously allocated.

Reply:
Please see https://github.com/atomvm/AtomVM/pull/418/commits/e324696c6f57752ce1e48bacb65f87a45b8c98a7

Reply:
OK. I am still not convinced we need to keep the current label, but ok.
However, you need to initialize the `current_label` above to the value from the passed `ctx` parameter.